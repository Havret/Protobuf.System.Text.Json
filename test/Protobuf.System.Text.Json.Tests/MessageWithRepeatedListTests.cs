using System.Text.Json;
using System.Text.Json.Protobuf.Tests;
using Protobuf.System.Text.Json.Tests.Utils;
using SmartAnalyzers.ApprovalTestsExtensions;
using Xunit;

namespace Protobuf.System.Text.Json.Tests;

public class MessageWithRepeatedListTests
{
    [Fact]
    public void Should_serialize_message_with_repeated_list()
    {
        // Arrange
        var msg = new MessageWithRepeatedList
        {
            Int32List = {1, 2, 3},
            Int64List = {int.MaxValue + 1L, int.MaxValue + 2L, int.MaxValue + 3L},
            EnumList = {Corpus.Images, Corpus.News},
            NestedMessageList =
            {
                new NestedField
                {
                    Field = 4
                }
            }
        };

        var jsonSerializerOptions = TestHelper.CreateJsonSerializerOptions();

        // Act
        var serialized = JsonSerializer.Serialize(msg, jsonSerializerOptions);

        // Assert
        var approver = new ExplicitApprover();
        approver.VerifyJson(serialized);
    }

    [Fact]
    public void Should_deserialize_message_with_repeated_list()
    {
        // Arrange
        var msg = new MessageWithRepeatedList
        {
            Int32List = {1, 2, 3},
            Int64List = {int.MaxValue + 1L, int.MaxValue + 2L, int.MaxValue + 3L},
            EnumList = {Corpus.Images, Corpus.News},
            NestedMessageList =
            {
                new NestedField
                {
                    Field = 4
                }
            }
        };

        var jsonSerializerOptions = TestHelper.CreateJsonSerializerOptions();

        // Act
        var serialized = JsonSerializer.Serialize(msg, jsonSerializerOptions);
        var deserialized = JsonSerializer.Deserialize<MessageWithRepeatedList>(serialized, jsonSerializerOptions);

        // Assert
        Assert.NotNull(deserialized);
        Assert.Equal(msg.Int32List, deserialized.Int32List);
        Assert.Equal(msg.Int64List, deserialized.Int64List);
        Assert.Equal(msg.EnumList, deserialized.EnumList);
        Assert.Equal(msg.NestedMessageList, deserialized.NestedMessageList);
    }
}