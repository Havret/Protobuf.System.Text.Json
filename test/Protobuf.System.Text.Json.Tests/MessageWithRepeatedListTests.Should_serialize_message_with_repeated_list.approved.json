{
  "int32List": [
    1,
    2,
    3
  ],
  "int64List": [
    2147483648,
    2147483649,
    2147483650
  ],
  "enumList": [
    2,
    4
  ]
}